linode_api4/objects/database.py

```diff
@@ -0,0 +1,405 @@
+    api_endpoint = '/databases/mysql/instances/{database_id}/backups/{id}'
+
+
+class MongoDBDatabaseBackup(DatabaseBackup):
+    api_endpoint = '/databases/mongodb/instances/{database_id}/backups/{id}'
+
+
+class PostgreSQLDatabaseBackup(DatabaseBackup):
+    api_endpoint = '/databases/postgresql/instances/{database_id}/backups/{id}'
+
+
+class MySQLDatabase(Base):
+    api_endpoint = '/databases/mysql/instances/{id}'
+
+    properties = {
+        'id': Property(identifier=True),
+        'label': Property(mutable=True, filterable=True),
+        'allow_list': Property(mutable=True),
+        'backups': Property(derived_class=MySQLDatabaseBackup),
+        'cluster_size': Property(),
+        'created': Property(is_datetime=True),
+        'encrypted': Property(),
+        'engine': Property(filterable=True),
+        'hosts': Property(),
+        'port': Property(),
+        'region': Property(filterable=True),
+        'replication_type': Property(),
+        'ssl_connection': Property(),
+        'status': Property(volatile=True, filterable=True),
+        'type': Property(filterable=True),
+        'updated': Property(volatile=True, is_datetime=True),
+        'updates': Property(mutable=True),
+        'version': Property(filterable=True),
+    }
+
+    @property
+    def credentials(self):
+        if not hasattr(self, '_credentials'):
+            resp = self._client.get('{}/credentials'.format(MySQLDatabase.api_endpoint), model=self)
+            self._set('_credentials', MappedObject(**resp))
+
+        return self._credentials
+
+    @property
+    def ssl(self):
+        if not hasattr(self, '_ssl'):
+            resp = self._client.get('{}/ssl'.format(MySQLDatabase.api_endpoint), model=self)
+            self._set('_ssl', MappedObject(**resp))
+
+        return self._ssl
+
+    def credentials_reset(self):
+        """
+        Reset the root password for a Managed MySQL Database.
+        """
+
+        self.invalidate()
+
+        return self._client.post('{}/credentials/reset'.format(MySQLDatabase.api_endpoint), model=self)
+
+    def patch(self):
+        """
+        Apply security patches and updates to the underlying operating system of the Managed MySQL Database.
+        """
+
+        self.invalidate()
+
+        return self._client.post('{}/patch'.format(MySQLDatabase.api_endpoint), model=self)
+
+    def backup_create(self, label, **kwargs):
+        """
+        Creates a snapshot backup of a Managed MySQL Database.
+        """
+
+        params = {
+            'label': label,
+        }
+        params.update(kwargs)
+
+        result = self._client.post('{}/backups'.format(MySQLDatabase.api_endpoint), model=self, data=params)
+        self.invalidate()
+
+        if 'id' not in result:
+            raise UnexpectedResponseError('Unexpected response when creating backup', json=result)
+
+        b = MySQLDatabaseBackup(self._client, result['id'], self.id, result)
+        return b
```

Comment:
It looks like creating a backup [doesn't return the new backup object](https://www.linode.com/docs/api/databases/#managed-mysql-database-backup-snapshot-create).  I'd argue that that's an API error, and I would request the API begin returning the new backup since that's how it works everywhere else.. but even so, this exception is always raised when successfully creating a backup right now.

This is true for all three database types at present.

Reply:
After discussing this issue, there doesn't seem to be a reliable solution to this at the moment from a client perspective.